src/context/PortfolioProvider/PortfolioContext.tsx

```diff
@@ -76,20 +76,36 @@ export const PortfolioProvider = ({ children }: { children: React.ReactNode }) =
   // once the wallet is connected, reach out to unchained to fetch
   // accounts for each chain/account specifier combination
   useEffect(() => {
-    if (isEmpty(accountSpecifiersList)) return
-    // clear the old portfolio, we have different non null data, we're switching wallet
-    console.info('dispatching portfolio clear action')
-    dispatch(portfolio.actions.clear())
+    const { getAccount } = portfolioApi.endpoints
+    // forceRefetch is enabled here to make sure that we always have the latest wallet information
+    // it also forces queryFn to run and that's needed for the wallet info to be dispatched
+    const options = { forceRefetch: true }
     // fetch each account
-    accountSpecifiersList.forEach(accountSpecifierMap => {
-      // forceRefetch is enabled here to make sure that we always have the latest wallet information
-      // it also forces queryFn to run and that's needed for the wallet info to be dispatched
-      dispatch(
-        portfolioApi.endpoints.getAccount.initiate({ accountSpecifierMap }, { forceRefetch: true }),
-      )
-    })
+    accountSpecifiersList.forEach(accountSpecifierMap =>
+      dispatch(getAccount.initiate({ accountSpecifierMap }, options)),
+    )
   }, [dispatch, accountSpecifiersList])
 
+  /**
+   * handle wallet disconnect/switch logic
+   */
+  useEffect(() => {
```

Comment:
I'm not sure that `PortfolioContext` is the right place for this sort of logic. The user could be looking at any page when the wallet changes. If everything in here happens regardless of which page is showing then it might be better to name it `AppContext` or something.

Is there an effect/dispatch that runs in `WalletProvider` when the wallet changes?

Reply:
useWallet returns a different wallet.

i agree, it's time to rename this to `AppContext`